x-pack/plugins/alerting/server/lib/snooze/is_snooze_active.ts

```diff
@@ -30,23 +57,32 @@ export function isSnoozeActive(snooze: RuleSnoozeSchedule) {
     };
 
   // Check to see if now is during a recurrence of the snooze
+
+  const { tzid, ...restRRule } = rRule;
+  const startDate = utcToLocalUtc(new Date(rRule.dtstart), tzid);
+  const nowDate = utcToLocalUtc(new Date(now), tzid);
+
   try {
     const rRuleOptions = {
-      ...rRule,
-      dtstart: new Date(rRule.dtstart),
-      until: rRule.until ? new Date(rRule.until) : null,
+      ...restRRule,
```

Comment:
The other important thing was we must stop passing `tzid` into RRule:

https://github.com/jakubroztocil/rrule/issues/550#issuecomment-1331369234

Reply:
It looks like removing `tzid` breaks days of the week. All days of the week will now be calculated relative to when the day starts in UTC.

Observe the snapshot you receive from modifying `'snooze is active byweekday'` to run at 6:15 on Feb 27 instead of 8:15 on Feb 25:
```
Snapshot name: `isSnoozeExpired snooze is active byweekday 1`

    - Snapshot  - 1
    + Received  + 1

      Object {
        "id": "9141dc1f-ed85-4656-91e4-119173105432",
    -   "lastOccurrence": 2023-02-24T23:00:00.000Z,
    +   "lastOccurrence": 2023-02-25T00:00:00.000Z,
        "snoozeEndTime": 2023-02-27T06:00:00.000Z,
      }
```

From what I can tell, it's generating **two** overlapping occurrences. One that starts at 11pm UTC on Friday the 24th, and then another that starts at midnight UTC on Saturday the 25th. Because it's Saturday in UTC now, on a new day, so that means it has to generate another recurrence.

Of course the reported `snoozeEndTime` is now inaccurate because the snooze doesn't actually end if you set the `fakeTimer` to that exact timestamp.

Reply:
Ah yes my bad, this is fixed now with the commit https://github.com/elastic/kibana/commit/d757a43980364d95bfda6c3d8ab0daabe5be9538

When I now run the test `'snooze is active byweekday'` with:

```
fakeTimer = sinon.useFakeTimers(new Date('2023-02-27T06:15:00.000Z'));
```
I get `null`, which seems to be correct as the snooze has ended.